test/ifElse/ifElseStep.js

```diff
@@ -368,4 +409,306 @@ describe('If Else step', () => {
+                            }
+                        ],
+                        type: "step"
+                    },
+                    if: {
+                        description: "If Step",
+                        steps: null,
+                        type: "step"
+                    },
+                    then: {
+                        description: "Then Step",
+                        steps: [
+                            {
+                                description: "nested step 1",
+                                steps: null,
+                                type: "step"
+                            },
+                            {
+                                description: "nested step 2",
+                                steps: null,
+                                type: "step"
+                            }
+                        ],
+                        type: "step"
+                    }
+                })
+            })
+
+            it('should audit', async () => {
+                //given
+                const st = givenAnIfThenStepUc()
+                //when
+                await st.run()
+                //then
+                assert.deepStrictEqual(st.auditTrail, {
+                    type: 'if else',
+                    description: undefined,
+                    returnIf: {
+                        description: "If Step",
+                        type: "step",
+                        elapsedTime: st.auditTrail.returnIf.elapsedTime,
+                        return: { Ok: true },
+                    },
+                    returnThen: {
+                        description: "Then Step",
+                        type: "step",
+                        elapsedTime: st.auditTrail.returnThen.elapsedTime,
+                        return: {
+                            Ok: {}
+                        },
+                        steps: [
+                            {
+                                description: "nested step 1",
+                                type: "step",
+                                elapsedTime: st.auditTrail.returnThen.steps[0].elapsedTime,
+                                return: {
+                                    Ok: "response nestedstep 1"
+                                }
+                            },
+                            {
+                                description: "nested step 2",
+                                type: "step",
+                                elapsedTime: st.auditTrail.returnThen.steps[1].elapsedTime,
+                                return: {
+                                    Ok: "response nestedstep 2"
+                                }
+                            }
+                        ]
+                    },
+                    elapsedTime: st.auditTrail.elapsedTime
+                })
+                assert.ok(st.auditTrail.elapsedTime > 0)
+            })
+        })
+
+        context('returning Ok from Else', () => {
```

Comment:
it is returning `Err`, not a `Ok` from `Then`

Reply:
fixed

Reply:
Shouldn't we have
`returning Ok from Then`
`returning Err from Then`
`returning Ok from Else`
`returning Err from Else` ?

Reply:
Sorry but I couldn't find the explanation why this was marked as resolved.

Reply:
oh sorry, I couldn't understand at that time, right now I fixed the correctly the test.